### PR TITLE
Feature/side drawer refactor

### PR DESCRIPTION
**Pull request by matthew-garrett**
- Refactors use-side-drawer-manager
- Refactors use-wallet-side-drawers
- Adds use-portfolio-side-drawers

### CODE REVIEW

File: packages/shared/config/src/enums.ts [hunk 1]
```diff
@@ -71,7 +71,32 @@ export enum PORTFOLIO_CATEGORIES {
...
 }
 
-export enum SIDEBAR_CATEGORIES {
+export enum SETTINGS_SIDE_DRAWERS {
+  SETTINGS = 'settings',
+  NOTIFICATIONS = 'notifications',
+  CONNECT_WALLET = 'connect-wallet',
+}
+
+export enum SIDE_DRAWERS {
```
Comment (jeffywu) [inline packages/shared/config/src/enums.ts:80]: Instead of copying and pasting these strings in two places, use a const instead for the SIDE_DRAWERS enum.
https://github.com/microsoft/TypeScript/issues/17592#issuecomment-320028196

```
enum BasicEvents {
    Start = "Start",
    Finish = "Finish"
};

// extend enum using "extends" keyword
const AdvEvents = {
    ...BasicEvents,
    Pause: "Pause",
    Resume: "Resume"
};
```

File: packages/shared/wallet/src/view-as-account/view-as-account.tsx [hunk 2]
```diff
@@ -38,11 +40,16 @@ export function ViewAsAccount() {
       }}
     >
       <Box
-        sx={{ borderTop: `1px solid ${theme.palette.borders.default}`, margin: '24px 0px' }}
+        sx={{
+          borderTop: `1px solid ${theme.palette.borders.default}`,
+          margin: theme.spacing(3, 0),
```
Comment (jeffywu) [inline packages/shared/wallet/src/view-as-account/view-as-account.tsx:45]: 👍🏽

**Review by jeffywu — APPROVED** [review]

File: .eslintrc.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
       "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
       "rules": {
         "@nrwl/nx/enforce-module-boundaries": [
-          "warn",
```
Comment (jeffywu) [inline .eslintrc.json:14]: @matthew-garrett just undo these changes so the build compiles, will have @mgnsharon look at the errors

File: packages/shared/config/src/enums.ts (hunk 1 above)
Comment (matthew-garrett) [reply packages/shared/config/src/enums.ts:80]: This does not work as expected
